spec/dd-agent_spec.rb

```diff
@@ -12,18 +24,22 @@
   it 'drops an agent config file' do
     expect(@chef_run).to create_template '/etc/dd-agent/datadog.conf'
   end
+end
 
-  it 'enables the datadog-agent service' do
-    expect(@chef_run).to enable_service 'datadog-agent'
+shared_examples_for 'common windows resources' do
+  it_behaves_like 'datadog-agent service'
+
+  it 'ensures the Datadog config directory exists' do
+    expect(@chef_run).to create_directory "#{ENV['ProgramData']}/Datadog"
```

Comment:
Instead of determining and interpolating ENV during the test, set it to something realistic in the block where you construct the Windows object, and then compare with the static value here.
This will eliminate needing to look up an environment variable during testing that is likely not going to be there anyhow.

See [here](https://gist.github.com/brettweavnet/6171363) for an example of how you might stub an environment variable.